core/src/main/scala/fs2/gzip/package.scala

```diff
@@ -126,28 +127,19 @@ package object gzip {
               }
 
               if (len > 0)
-                Stream.chunk(Chunk.bytes(inner, 0, len)).covary[F]
-              else
-                Stream.empty[F]
-            }
-          }
-
-          // drains abis through gzis
-          @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
-          lazy val repeatedlyDecompress: Stream[F, Byte] = Stream force {
-            Sync[F] delay {
-              if (abis.available() > 0)
-                stepDecompress ++ repeatedlyDecompress
+                Stream.chunk(Chunk.bytes(inner, 0, len)).covary[F] ++ stepDecompress
               else
                 Stream.empty[F]
             }
           }
 
+          // TODO: It is possible for this to fail with a non-progressive error
+          //       if `in` contains bytes in addition to the compressed data.
```

Comment:
Wouldn't that be an invalid bytestream? I'm not sure I understand this comment.

Reply:
It would, but the error would be misleading.

Reply:
Ah yeah, I definitely want to improve the errors. There's also a more specific error we can give for header-related issues.